tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -0,0 +1,284 @@
+      const expectedEntries: [
+        Uri,
+        Diagnostic[] | undefined
+      ][] = prepDiagnostics();
+      await testUpdateMtaDiagnostics(expectedEntries);
+    });
+
+    async function testUpdateMtaDiagnostics(
+      expectedCollectionEntries: [Uri, Diagnostic[] | undefined][]
+    ) {
+      const fileName = "mtaValidations.spec.js";
+      const uriPath = resolve(__dirname, fileName);
+      const uri = {
+        fsPath: uriPath,
+      } as Uri;
+
+      let collectionEntries:
+        | [Uri, Diagnostic[] | undefined][]
+        | undefined = undefined;
+      sinon.stub(testVscode.languages, "createDiagnosticCollection").returns({
+        forEach: () => {
+          return;
+        },
+        set: (newCollectionEntries: [Uri, Diagnostic[] | undefined][]) => {
+          collectionEntries = newCollectionEntries;
+        },
+      });
+
+      await updateMtaDiagnostics(uri, disposables);
+      expect(collectionEntries).to.deep.equal(expectedCollectionEntries);
+    }
+  });
+  describe("validateWsMtaYamls", async () => {
+    it("returns no diagnostics when mta.yaml does not exist in the WS", async () => {
+      sinon
+        .stub(testVscode.workspace, "findFiles")
+        .returns(Promise.resolve([]));
+      const createDiagnosticCollectionSpy = sinon.stub(
+        testVscode.languages,
+        "createDiagnosticCollection"
+      );
+
+      await validateWsMtaYamls(disposables);
+
+      expect(createDiagnosticCollectionSpy.callCount).to.equal(0);
+    });
+
+    it("returns no diagnostics when mta.yaml and dev.mtaext have no errors in the WS", async () => {
+      sinon.stub(fsExtra, "pathExists").resolves(true);
+      sinon.stub(Mta.prototype, "validate").resolves({});
+
+      await testValidateWsMtaYamls([]);
+    });
+
+    it("returns diagnostics when mta.yaml and dev.mtaext have errors in the WS", async () => {
+      const expectedEntries: [
+        Uri,
+        Diagnostic[] | undefined
+      ][] = prepDiagnostics();
+      await testValidateWsMtaYamls(expectedEntries);
+    });
+
+    async function testValidateWsMtaYamls(
+      expectedCollectionEntries: [Uri, Diagnostic[] | undefined][]
+    ) {
+      const fileName = "mtaValidations.spec.js";
+      const uriPath = resolve(__dirname, fileName);
+      const uri = {
+        fsPath: uriPath,
+      } as Uri;
+      sinon
+        .stub(testVscode.workspace, "findFiles")
+        .returns(Promise.resolve([uri]));
+
+      let collectionEntries:
+        | [Uri, Diagnostic[] | undefined][]
+        | undefined = undefined;
+      sinon.stub(testVscode.languages, "createDiagnosticCollection").returns({
+        forEach: () => {
+          return;
+        },
+        set: (newCollectionEntries: [Uri, Diagnostic[] | undefined][]) => {
+          collectionEntries = newCollectionEntries;
+        },
+      });
+
+      await validateWsMtaYamls(disposables);
+      expect(collectionEntries).to.deep.equal(expectedCollectionEntries);
+    }
```

Comment:
You should clear the diagnostics collection cache `afterEach`

Reply:
done